tools/compiletest/src/runtest.rs

```diff
@@ -310,6 +324,57 @@ impl<'test> TestCx<'test> {
         self.compose_and_run(kani)
     }
 
+    /// Runs an executable file and:
+    ///  * Checks the expected output if an expected file is specified
+    ///  * Checks the exit code (assumed to be 0 by default)
+    fn run_exec_test(&self) {
+        // Open the `config.yml` file and extract its values
+        let path_yml = self.testpaths.file.join("config.yml");
+        let config_file = std::fs::File::open(path_yml).expect("couldn't open `config.yml`");
+        let exec_config: ExecConfig =
+            serde_yaml::from_reader(config_file).expect("couldn't read `config.yml` values");
+
+        // Check if the `script` file exists
+        let script_rel_path = PathBuf::from(exec_config.script);
+        let script_path = self.testpaths.file.join(script_rel_path);
+        if !script_path.exists() {
+            let err_msg = format!("test failed: couldn't find script in {}", script_path.display());
+            self.error(&err_msg);
+        }
+
+        // Check if the `expected` file exists, and load its contents into `expected_output`
+        let expected_output = if let Some(expected_path) = exec_config.expected {
+            let expected_rel_path = PathBuf::from(expected_path);
+            let expected_path = self.testpaths.file.join(expected_rel_path);
+            if !expected_path.exists() {
+                let err_msg = format!(
+                    "test failed: couldn't find expected file in {}",
+                    expected_path.display()
+                );
+                self.error(&err_msg);
+            }
+            Some(fs::read_to_string(expected_path).unwrap())
+        } else {
+            None
+        };
+
+        // Create the command `sh script` and run it from the test directory
+        let mut script_path_cmd = Command::new("time");
+        script_path_cmd.arg(script_path).current_dir(&self.testpaths.file);
+        let proc_res = self.compose_and_run(script_path_cmd);
+
+        // Compare with expected output if it was provided
+        if let Some(output) = expected_output {
+            self.verify_output(&proc_res, &output);
+        }
+
+        // Compare with exit code (0 if it wasn't provided)
+        let expected_code = exec_config.code.or(Some(0));
+        if proc_res.status.code() != expected_code {
+            self.fatal_proc_rec("test failed: expected different code", &proc_res);
```

Comment:
Can we add the expected code and the actual code to the message?

Reply:
Done.